### ISSUE TITLE
Is it compatible with mod added armor?

### ISSUE THREAD

**Issue by ardissaps** [Fourmisain/DyeAllTheThings#1]

**Comment by Fourmisain** [comment]
Yes, I do mean **all** armor!
In the screenshots of the mod page I'm actually wearing some BetterEnd and MC Dungeons Armor.

Lemme know if you find any armor that isn't dyable!

**Comment by ardissaps** [comment]
It worked on most armor, even worked with [Cauldron Dyeing](https://modrinth.com/mod/cauldron-dyeing), which is really cool! 
![image](https://user-images.githubusercontent.com/82106625/175776740-15bd1766-8f74-4aab-952e-360fff51b443.png)

But,
@Fourmisain it works wonderfull except for [Immersive Armor](https://www.curseforge.com/minecraft/mc-mods/immersive-armors), the inventory model changed but on body still original colour.
![image](https://user-images.githubusercontent.com/82106625/175776787-472e0060-efb8-4f09-8821-1f7009c0f0aa.png)

**Comment by Fourmisain** [comment]
Oh I see, Immersive Armor uses some [custom rendering](https://github.com/Luke100000/ImmersiveArmors/blob/1.18.2/common/src/main/java/immersive_armors/mixin/MixinArmorFeatureRenderer.java), bypassing the vanilla rendering.

The good news is that Immersive Armor does have dyeable armor too and I found a way to use this and add compatibility.

Just released a new version, CurseForge should soon approve it - or get it from Modrinth now.

Note I only tested it with the latest version (1.3.5) of Immersive Armors for 1.18.2.
